### PR TITLE
removed check for valid keys from contexts

### PR DESCRIPTION
**Pull request by mohammadkolahdouzan**

### CODE REVIEW

File: src/main/java/org/mondemand/Client.java [hunk 1]
```diff
@@ -344,12 +344,9 @@ public void addContext(String key, String value) throws MondemandException {
     if(key == null) {
       throw new MondemandException("key is null");
     }
-    if(value == null || value.isEmpty()) {
+    if(value == null) {
       throw new MondemandException("value is empty or null");
```
Comment (kkharma) [inline src/main/java/org/mondemand/Client.java]: we are ok with empty sting keys?  if so, the exception message should be updated.

Comment (kkharma) [reply src/main/java/org/mondemand/Client.java]: duh sorry it's a value, not a key... but exception message should still be updated.

Comment (mohammadkolahdouzan) [reply src/main/java/org/mondemand/Client.java]: I was going to ask whether or not we should be ok with `null` key/values too, or not?

File: src/main/java/org/mondemand/Client.java [hunk 2]
```diff
@@ -344,12 +344,9 @@ public void addContext(String key, String value) throws MondemandException {
...
     }
-    if(!isKeyValid(key)) {
-      throw new MondemandException("key is invalid: " + key);
-    }
 
     Context ctxt = new Context(key, value);
```
[the remark this answers is not in the document]
Comment (mohammadkolahdouzan) [reply src/main/java/org/mondemand/Client.java]: nice catch. I removed it from the constructor too.

File: src/main/java/org/mondemand/Context.java [hunk 3]
```diff
@@ -35,18 +33,9 @@ public String getValue() {
...
     value = v;
-    if(key == null) {
-      throw new MondemandException("key is null");
-    }
-    if(value == null || value.isEmpty()) {
-      throw new MondemandException("value is empty or null");
-    }
-    if(!isKeyValid(key)) {
-      throw new MondemandException("key is invalid: " + key);
-    }
```
Comment (kkharma) [inline src/main/java/org/mondemand/Context.java]: would removing this make it possible to get a bad context in if addContext were bypassed?

Comment (mohammadkolahdouzan) [reply src/main/java/org/mondemand/Context.java]: what is a bad context though? only non-empty key/values should be ok?

Comment (mohammadkolahdouzan) [reply src/main/java/org/mondemand/Context.java]: I can put the checks for `null` back and only drop the one for `isKeyValid()` if that is the right thing to do.

Comment (nimaco) [reply src/main/java/org/mondemand/Context.java]: yeah, I think null checks are enough.  you can remove the redundant null checks from addContext(), right?

Comment (kkharma) [reply src/main/java/org/mondemand/Context.java]: yeah what nick said

Comment (mohammadkolahdouzan) [reply src/main/java/org/mondemand/Context.java]: removed null check from `addContext`, only kept the `null` checks in constructor of `Context` and removed the `isKeyValid()` from it.

**Comment by nimaco** [comment]
lgtmtoo

**Comment by mohammadkolahdouzan** [comment]
thanks Nick and Ken for the review.

**Pull request merged**